scripts/main.py

```diff
@@ -8,21 +8,32 @@
 
 
 def run(config: dict, cache: str):
+    # Get uptake data from the cache
     data = nisapi.get_nis(cache)
 
-    # List of the cumulative data sets described in the yaml
-    # note that df.filter(**{"column1": value1, "column2": value2}) is equivalent to
-    # df.filter(pl.col("column1")==value1, pl.col("column2")==value2)
+    # Prune data to correct rows and columns
     cumulative_data = [
-        iup.CumulativeUptakeData(data.filter(**x["filters"]).collect())
+        iup.CumulativeUptakeData(
+            data.filter(**x["filters"])
+            .collect()
+            .rename(x["keep"])
+            .select(x["keep"].values())
+            .sort("date")
+        )
         for x in config["data"].values()
     ]
 
-    # List of grouping factors used in each data set
+    # Find grouping factors common to all data sets
```

Comment:
I'm realizing that "factor" might have confused me

E.g., we have groups of concepts like: geographic type ("state") and geography ("Wyoming"). Is Wyoming the factor, or state?

Reply:
"State" is the factor, and "Wyoming" is a level of that factor. `# Find grouping factors common to all data sets` is confusing language - what I'm trying to say is: "make sure the same columns for grouping factors are present in every data set that will contribute to the model." I will rephrase.